spec/Section 6 -- Execution.md

```diff
@@ -359,17 +348,212 @@ Unsubscribe(responseStream):
 
 ## Executing Selection Sets
 
-To execute a _selection set_, the object value being evaluated and the object
-type need to be known, as well as whether it must be executed serially, or may
-be executed in parallel.
+The process of executing a GraphQL operation is to recursively execute every
+selected field in the operation. To do this, first all initially selected fields
+from the operation's top most _root selection set_ are collected, then each
+executed, then of those all subfields are collected, then each executed. This
+process continues until there are no more subfields to collect and execute.
+
+### Executing the Root Selection Set
+
+:: A _root selection set_ is the top level _selection set_ provided by a GraphQL
+operation. A root selection set always selects from a root type.
 
-First, the selection set is turned into a grouped field set; then, each
-represented field in the grouped field set produces an entry into a result map.
+To execute the root selection set, the initial value being evaluated and the
+root type must be known, as well as whether it must be executed serially, or may
+be executed in parallel (see
+[Normal and Serial Execution](#sec-Normal-and-Serial-Execution).
 
-ExecuteSelectionSet(selectionSet, objectType, objectValue, variableValues):
+Executing the root selection set works similarly for queries (parallel),
+mutations (serial), and subscriptions (where it is executed for each event in
+the underlying Source Stream).
+
+First, the _selection set_ is collected into a _grouped field set_ which is then
+executed, returning the resulting {data} and {errors}.
+
+ExecuteRootSelectionSet(variableValues, initialValue, objectType, selectionSet,
+executionMode):
 
 - Let {groupedFieldSet} be the result of {CollectFields(objectType,
   selectionSet, variableValues)}.
+- Let {data} be the result of running {ExecuteGroupedFieldSet(groupedFieldSet,
+  objectType, initialValue, variableValues)} _serially_ if {executionMode} is
+  {"serial"}, otherwise _normally_).
```

Comment:
I think we should restore the parenthesis, they add clarity to what "normally" means.

```suggestion
  {"serial"}, otherwise _normally_ (allowing parallelization)).
```